src/lib/comms/sol-socket-dtls-impl-tinydtls.c

```diff
@@ -699,54 +675,58 @@ get_psk_info(struct dtls_context_t *ctx, const session_t *session,
 }
 
 struct sol_socket *
-sol_socket_dtls_wrap_socket(struct sol_socket *to_wrap)
+sol_socket_default_dtls_new(const struct sol_socket_default_options *options)
 {
-    static const struct sol_socket_impl impl = {
+    static const struct sol_socket_type type = {
         .bind = sol_socket_dtls_bind,
         .join_group = sol_socket_dtls_join_group,
         .sendmsg = sol_socket_dtls_sendmsg,
         .recvmsg = sol_socket_dtls_recvmsg,
-        .set_on_write = sol_socket_dtls_set_on_write,
-        .set_on_read = sol_socket_dtls_set_on_read,
-        .del = sol_socket_dtls_del,
-        .new = NULL,
-        .setsockopt = sol_socket_dtls_setsockopt,
-        .getsockopt = sol_socket_dtls_getsockopt
+        .set_write_monitor = sol_socket_dtls_set_write_monitor,
+        .set_read_monitor = sol_socket_dtls_set_read_monitor,
+        .del = sol_socket_dtls_del
     };
     static const dtls_handler_t dtls_handler = {
         .write = write_encrypted,
         .read = call_user_read_cb,
         .event = handle_dtls_event,
         .get_psk_info = get_psk_info
     };
-    struct sol_socket_dtls *socket;
+    struct sol_socket_default_options opts = *options;
+    struct sol_socket_dtls *s;
 
     init_dtls_if_needed();
+    opts.on_can_read = read_encrypted;
+    opts.on_can_write = call_user_write_cb;
```

Comment:
this whole way of doing the dtls wrapping seems strange, you can leave this in but review it to be more natural later.